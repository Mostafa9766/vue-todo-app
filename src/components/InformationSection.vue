<template>
  <div>
    <div>
      <el-row type="flex" justify="center" class="counterSection">
        <el-col :span="4">
          Completed :
        </el-col>
        <el-col :span="2">
          {{completed}}
        </el-col>
        <el-col :span="3">
          Total :
        </el-col>
        <el-col :span="2">
          {{total}}
        </el-col>
      </el-row>
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex';

export default {
  name: 'InformationSection',
  computed: {
    ...mapState(['activities']),
    completed() {
      return this.activities.filter((val) => val.completed === true).length;
    },
    total() {
      return this.activities.length;
    },
  },
};
</script>

<style>

.counterSection {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}
</style>
