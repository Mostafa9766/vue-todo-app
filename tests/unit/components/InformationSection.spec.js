import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import ElementUi from 'element-ui';
import InformationSection from '@/components/InformationSection.vue';

describe('InformationSection.vue', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);
  localVue.use(ElementUi);

  let store;
  let state;

  beforeEach(() => {
    state = {
      activities: [],
    };

    store = new Vuex.Store({
      state,
    });
  });

  it('render InformationSection component', () => {
    const wrapper = shallowMount(InformationSection, {
      localVue,
      store,
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toBeTruthy();
  });
});
