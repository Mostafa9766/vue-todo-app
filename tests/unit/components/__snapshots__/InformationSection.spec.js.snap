// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationSection.vue render InformationSection component 1`] = `
<div>
  <div>
    <el-row-stub tag="div" type="flex" justify="center" class="counterSection">
      <el-col-stub span="4" tag="div">
        Completed :
      </el-col-stub>
      <el-col-stub span="2" tag="div">
        0
      </el-col-stub>
      <el-col-stub span="3" tag="div">
        Total :
      </el-col-stub>
      <el-col-stub span="2" tag="div">
        0
      </el-col-stub>
    </el-row-stub>
  </div>
</div>
`;
